uipyqt.py

```diff
@@ -0,0 +1,288 @@
+                table.append(table_btn.text())
+                table_card_set=True
+
+        sum_of_table=tactics.sum_of_cards(table)
+
+        hand_card_set=False
+        hand_card_value=-1
+        for hand_btn in self.my_hand_buttons:
+            if hand_btn.isChecked():
+                hand_card_set=True
+                hand_card_value=int(hand_btn.text()[:2])
+
+        self.lay_card_button.setEnabled(hand_card_set and not table_card_set)
+
+        self.claim_cards_button.setEnabled(hand_card_set and sum_of_table == hand_card_value)
+
+    # again human move
+    def pressed_OK(self):
+        self.my_move()
+
+    # remove card from hand and add it to the table
+    def pressed_lay_card(self):
+        for button in self.my_hand_buttons:
+            if button.isChecked():
+                laid_card = button.text()
+                self.sc.hands[0].remove(laid_card)
+                self.sc.table.append(laid_card)
+                break
+
+        # now it's computer's move
+        self.opponent_move()
+
+    # remove card from hand, cards from the table - and add them to pile
+    def pressed_claim_cards(self):
+        for button2 in self.my_hand_buttons:
+            if button2.isChecked():
+                card_from_hand=button2.text()
+                self.sc.hands[0].remove(card_from_hand)
+                self.sc.piles[0].append(card_from_hand)
+                break
+
+        for button in self.table_buttons:
+            if button.isChecked():
+                card_from_table=button.text()
+                self.sc.table.remove(card_from_table)
+                self.sc.piles[0].append(card_from_table)
+
+        # if table is empty, calculate scopa
+        if len(self.sc.table) == 0:
+            self.sc.scopa_count[0] += 1
+            qmsg = QMessageBox()
+            qmsg.setWindowTitle("Scopa")
+            qmsg.setText(tactics.this_was_scopa)
+            qmsg.exec_()
+
+        self.last_claimed_hand = 0
+        self.opponent_move()
+
+    # if no cards in game claim remaining cards for the player who made the last claim
+    def finish_game(self):
+        for card in self.sc.table.copy():
+            self.sc.piles[self.last_claimed_hand].append(card)
+            self.sc.table.remove(card)
+
+        my_score = self.sc.calculate_score(0)
+        opponent_score = self.sc.calculate_score(1)
+
+        message = "Moje karty: " + str(len(self.sc.piles[0])) + ", w tym denarów "
+        message += str(tactics.denars(self.sc.piles[0])) + ", wynik: " + str(my_score)
+        message += "\nPrzeciwnika: " + str(len(self.sc.piles[1])) + ", w tym denarów "
+        message += str(tactics.denars(self.sc.piles[1])) + " wynik: " + str(opponent_score)
+
+
+        msg = QMessageBox()
+        msg.setText("Scopa finished")
+        msg.setInformativeText(message)
+        msg.setStandardButtons(QMessageBox.Ok)
+        msg.exec_()
+
+        # self.sc.print_game_results()
+        sys.exit(0)
```

Comment:
Try to separate this method.

Reply:
in other words, divide this method's logic into smaller units